### PR TITLE
Adds tests to check when clicking into an anchor element within SVG does not throw any errors. | Fixes #274

### LINKED ISSUE #274: Senna doesn't navigate correctly using a <a /> tag inside a <svg /> tag
### Expected behaviour

There is a `<a />` tag inside SVG and I hope that when I click the link, Senna can load the page in SPA.

### Actual behaviour

Clicking on the link displays a javascript error in the console:
screen shot 2018-08-29 at 19 28 31

This happens because the type of the `<a />` tag inside an SVG is different from the `<a />` tag inside HTML and Senna expects a string instead of an `SVGAnimatedString` object:
screen shot 2018-08-29 at 19 27 20

### Steps to reproduce

1. Add a `<a />` tag inside an SVG:
screen shot 2018-08-29 at 19 33 27
2. Click on the link to navigate with SPA;

### Browsers affected

Shouldn't be working on all browsers

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

File: src/app/App.js [hunk 1]
```diff
@@ -943,7 +943,18 @@ class App extends EventEmitter {
 			console.log('Navigate aborted, invalid mouse button or modifier key pressed.');
 			return;
 		}
-		this.maybeNavigate_(event.delegateTarget.href, event);
+
+		let href = event.delegateTarget.href;
+
+		// If SVGAnimatedString object is passed as url
+		if (isObject(href) && href.toString() === '[object SVGAnimatedString]') {
```
Comment (jbalsas) [inline src/app/App.js]: Hey @diegonvs, this looks overly specific and way too complicated... isn't there a better way to deal with this? Can't we query the svg dom in a more meaningful way?

**Comment by diegonvs** [comment]
Hey @jbalsas, I'd researched some SVG specifications and there is no way to work around this. So, I've created a function that treat this hyperlink references. About the function that creates an element to get absolute `href`: this work around is necessary because we cannot pass relative paths to maybeNavigate_ function and a treatment using metal-uri is more complex. Do you have any thought about it?

File: src/utils/utils.js [hunk 2]
```diff
@@ -84,6 +99,26 @@ class utils {
 		return uri.getPathname();
 	}
 
+	/**
+	 * Receives a event and returns the delegate target href.
+	 * @param {!Event} event
+	 * @return {string}
+	 */
+	static getHref(event) {
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply src/utils/utils.js]: Hey @jbalsas , I've made this changes to make code more comprehensible when using this function. However, receiving a Event object is unnecessary. I'll make a correction.

File: src/utils/utils.js [hunk 3]
```diff
@@ -84,6 +99,26 @@ class utils {
...
+		let href = event.delegateTarget.href;
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply src/utils/utils.js]: I had made this change thinking if we have a navigation using an anchor within SVG when scheduling a navigation

[the remark this answers is not in the document]
Comment (diegonvs) [reply src/utils/utils.js]: Reviewing the behaviour when scheduling a navigation I think that is unnecessary. I'll remove :)

**Comment by diegonvs** [comment]
Hey @jbalsas could you rerun travis?

**Comment by jbalsas** [comment]
Seems to be consistently failing, @diegonvs. Retrying one more time.

**Comment by jbalsas** [comment]
I'm pretty sure we won't be merging this for now, so going to close as I triage open PRs ;)

**Pull request closed without merging**